src/lib/components/chat/MessagesVerifier.svelte

```diff
@@ -0,0 +1,254 @@
+						<p class="text-xs text-gray-500 dark:text-gray-400">ID: {message.chatCompletionId}</p>
+					</div>
+				</div>
+			{/each}
+
+			<!-- Signature Details Section -->
+			<div class="space-y-3">
+				<div class="flex justify-between items-center">
+					<h3 class="text-sm font-medium text-gray-900 dark:text-white mt-4">Signature Details</h3>
+					<!-- <span class="text-xs text-gray-500 dark:text-gray-400">
+						{Object.keys(signatures).length} of {chatCompletions.length} signatures loaded
+					</span> -->
+				</div>
+				
+				{#if selectedMessageId && signatures[selectedMessageId]}
+					<!-- Verify on Etherscan Link -->
+					{#if signatures[selectedMessageId].signature}
+						<a class="mb-3" href="https://etherscan.io/verifiedSignatures" target="_blank">
+							<button
+								class="flex items-center text-blue-600 hover:text-blue-700 text-xs transition-colors"
+							>
+								<svg class="w-3 h-3 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24">
+									<path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14" />
+								</svg>
+								Verify the signature on Etherscan
+							</button>
+						</a>
+					{/if}
+
+					<div class="border border-gray-200 dark:border-gray-700 rounded-lg p-3 min-h-[150px]">
+						<!-- Signing Address -->
+						<div class="mb-2">
+							<label class="block text-xs font-medium text-gray-700 dark:text-gray-300 mb-1">Signing Address:</label>
+							<div class="px-2 py-1 bg-gray-100 dark:bg-gray-700 border border-gray-300 dark:border-gray-600 rounded text-xs font-mono break-all min-h-[24px] flex items-center">
+								{signatures[selectedMessageId].signing_address ?? ''}
+							</div>
+						</div>
+
+						<!-- Message Hash -->
+						<div class="mb-2">
+							<label class="block text-xs font-medium text-gray-700 dark:text-gray-300 mb-1">Message:</label>
+							<div class="px-2 py-1 bg-gray-100 dark:bg-gray-700 border border-gray-300 dark:border-gray-600 rounded text-xs font-mono break-all min-h-[24px] flex items-center">
+								{signatures[selectedMessageId].text ?? ''}
+							</div>
+						</div>
+
+						<!-- Signature -->
+						<div class="mb-2">
+							<label class="block text-xs font-medium text-gray-700 dark:text-gray-300 mb-1">Signature:</label>
+							<div class="px-2 py-1 bg-gray-100 dark:bg-gray-700 border border-gray-300 dark:border-gray-600 rounded text-xs font-mono break-all min-h-[24px] flex items-center">
+								{signatures[selectedMessageId].signature ?? ''}
+							</div>
+						</div>
+
+						<!-- Algorithm -->
+						<div>
+							<label class="block text-xs font-medium text-gray-700 dark:text-gray-300 mb-1">Algorithm:</label>
+							<div class="px-2 py-1 bg-gray-100 dark:bg-gray-700 border border-gray-300 dark:border-gray-600 rounded text-xs min-h-[24px] flex items-center">
+								{signatures[selectedMessageId].signing_algo || 'ecdsa'}
```

Comment:
Default algo here is meaningless, I suggest to use empty string or just hide the modal